tests/system/src/test/java/test/robot/javafx/scene/dialog/DialogWithOwnerSizingTest.java

```diff
@@ -0,0 +1,161 @@
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.WindowEvent;
+import javafx.scene.input.MouseButton;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+//see JDK8193502
+public class DialogWithOwnerSizingTest {
+    static Robot robot;
+    static Button button;
+    static Stage stage;
+    static Scene scene;
+    static Dialog<ButtonType> dialog;
+    static Dialog<ButtonType> dialog2;
+    static CountDownLatch startupLatch;
+    static CountDownLatch dialogShownLatch;
+    static CountDownLatch dialogHideLatch;
+
+    @Test(timeout = 15000)
+    public void dialogWithOwnerSizingTest() throws Exception {
+        Thread.sleep(500);
+        clickButton();
+        Thread.sleep(500);
+        Assert.assertEquals(dialog.getDialogPane().getWidth(), dialog2.getDialogPane().getWidth(), 0.0);
+        Assert.assertEquals(dialog.getDialogPane().getHeight(), dialog2.getDialogPane().getHeight(), 0.0);
+        hide();
+    }
+
+
+    private void clickButton() throws Exception {
+        dialogShownLatch = new CountDownLatch(2);
+        mouseClick(button.getLayoutX() + button.getWidth() / 2, button.getLayoutY() + button.getHeight() / 2);
+
+        Thread.sleep(400);
+        waitForLatch(dialogShownLatch, 10, "Failed to show Dialog");
+    }
+
+    private void hide() throws Exception {
+        dialogHideLatch = new CountDownLatch(2);
+        Platform.runLater(() -> {
+            dialog.close();
+            dialog2.close();
+        });
+        Thread.sleep(600);
+        waitForLatch(dialogHideLatch, 10, "Failed to hide Dialog");
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        new Thread(() -> Application.launch(SizingTestApp.class, (String[]) null)).start();
+        waitForLatch(startupLatch, 15, "FX runtime failed to start.");
+    }
+
+    @AfterClass
+    public static void exit() {
+        Platform.runLater(() -> stage.hide());
+        Platform.exit();
+    }
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                    (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    public static class SizingTestApp extends Application {
+        @Override
+        public void start(Stage primaryStage) {
+            robot = new Robot();
+            stage = primaryStage;
+
+            button = new Button("Open Dialogs");
+
+            scene = new Scene(button, 200, 200);
+            stage.setScene(scene);
+
+            stage.initStyle(StageStyle.UNDECORATED);
```

Comment:
Since this is a Robot test that relies on the Stage being in front of other windows, can you add the following?

```
    stage.setAlwaysOnTop(true);
```

Reply:
ok